mypy/binder.py

```diff
@@ -36,6 +37,11 @@ class CurrentType(NamedTuple):
     from_assignment: bool
 
 
+class UnreachableType(enum.Enum):
```

Comment:
As part of https://github.com/python/mypy/pull/18707 I've been thinking about related concepts, and I think it's possible to do this without an enum actually: simply mark skipped blocks as suppressed (currently they're only marked as unreachable) and if all the blocks flowing into a block are unreachable, check if any of them are suppressed -- if so, then the current block is suppressed.